front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -311,6 +341,77 @@ export function AgentMessage({
       }
     );
   }
+  function Citations({
+    activeReferences,
+  }: {
+    activeReferences: { index: number; document: RetrievalDocumentType }[];
+  }) {
+    const citationContainer = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      if (citationContainer.current && lastHoveredReference !== null) {
+        // get the offset of the first child in its container
+        const offset = (
+          citationContainer.current.firstElementChild
+            ?.firstElementChild as HTMLElement
+        ).offsetLeft;
+        const scrolling =
+          (citationContainer.current.firstElementChild?.firstElementChild
+            ?.scrollWidth || 0) *
+          (lastHoveredReference - 2);
+        citationContainer.current.scrollLeft = scrolling - offset;
```

Comment:
let scroll smoothly? I think we have example in the scroller of the conversation

Reply:
This was a personal preference, I don't like waiting for the transitions. Loosely held, so will change to smooth scroll